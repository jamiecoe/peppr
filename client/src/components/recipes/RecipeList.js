import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import _ from 'lodash';
import Navbar from '../Navbar';

class RecipeList extends Component {
  extraPosts() {
    const remainder = _.size(this.props.recipes) % 3;

    if (remainder) {
      const remainderArray = [];
      let i = 0;
      while (i < 3 - remainder) {
        remainderArray.push(<div className="recipeList__link" key={i} />);
        i += 1;
      }
      return remainderArray;
    }
    return null;
  }

  renderPosts() {
    return _.map(this.props.recipes, recipe => (
      <Link
        to={`/recipes/${recipe.id}`}
        className="recipeList__link recipelist--overlay"
        key={recipe.id}
      >
        <div
          className="recipeList__link--container"
          style={{
            backgroundImage: `linear-gradient(rgba(255,255,255,0.1), rgba(255,255,255,0.1)), url(${
              recipe.imageurl
            })`,
          }}
        >
          <div className="recipeList__link--innercontainer">
            <h2 className="recipeList__recipeTitle">{recipe.title}</h2>
          </div>
        </div>
      </Link>
    ));
  }

  render() {
    return (
      <div className="recipeList__container">
        <div className="recipeList__allrecipes">
          <div className="recipeList__link">
            <h1 className="recipeList__title">My Recipes</h1>
          </div>
          <div className="recipeList__link" />
          <div className="recipeList__link recipeList__hidden-2x2" />
          {this.renderPosts()}
          {this.extraPosts()}
        </div>
        <Navbar />
      </div>
    );
  }
}

const mapStateToProps = ({ recipes }) => ({ recipes });

export default connect(mapStateToProps)(RecipeList);
