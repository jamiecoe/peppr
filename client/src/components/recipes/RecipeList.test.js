import React from 'react';
import { shallow } from 'enzyme';
import { RecipeList } from './RecipeList';

describe('RecipeList', () => {
  const mockRecipes = [
    { id: 1, imageurl: '', title: '' },
    { id: 2, imageurl: '', title: '' },
  ];
  const props = { recipes: mockRecipes };
  const recipeList = shallow(<RecipeList {...props} />);

  it('renders properly', () => {
    expect(recipeList).toMatchSnapshot();
  });

  it('should render a container for each recipe in props', () => {
    expect(recipeList.find('.recipeList__link--container')).toHaveLength(mockRecipes.length);
  });

  it('should render enough .recipeList__link containers to always fill 3 columns', () => {
    expect(recipeList.find('.recipeList__link').length % 3).toBe(0);
  });
});
