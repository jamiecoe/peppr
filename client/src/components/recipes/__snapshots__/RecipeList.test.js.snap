// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeList renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RecipeList
    recipes={
        Array [
            Object {
              "id": 1,
              "imageurl": "",
              "title": "",
            },
            Object {
              "id": 2,
              "imageurl": "",
              "title": "",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="recipeList__allrecipes"
>
          <div
                    className="recipeList__link"
          >
                    <h1
                              className="recipeList__title"
                    >
                              My Recipes
                    </h1>
          </div>
          <div
                    className="recipeList__link"
          />
          <div
                    className="recipeList__link recipeList__hidden-2x2"
          />
          <Link
                    className="recipeList__link recipelist--overlay"
                    replace={false}
                    to="/recipes/1"
          >
                    <div
                              className="recipeList__link--container"
                              style={
                                        Object {
                                                  "backgroundImage": "linear-gradient(rgba(255,255,255,0.1), rgba(255,255,255,0.1)), url()",
                                                }
                              }
                    >
                              <div
                                        className="recipeList__link--innercontainer"
                              >
                                        <h2
                                                  className="recipeList__recipeTitle"
                                        />
                              </div>
                    </div>
          </Link>
          <Link
                    className="recipeList__link recipelist--overlay"
                    replace={false}
                    to="/recipes/2"
          >
                    <div
                              className="recipeList__link--container"
                              style={
                                        Object {
                                                  "backgroundImage": "linear-gradient(rgba(255,255,255,0.1), rgba(255,255,255,0.1)), url()",
                                                }
                              }
                    >
                              <div
                                        className="recipeList__link--innercontainer"
                              >
                                        <h2
                                                  className="recipeList__recipeTitle"
                                        />
                              </div>
                    </div>
          </Link>
          <div
                    className="recipeList__link"
          />
</div>,
        <Connect(Navbar) />,
      ],
      "className": "recipeList__container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="recipeList__link"
>
              <h1
                            className="recipeList__title"
              >
                            My Recipes
              </h1>
</div>,
            <div
              className="recipeList__link"
/>,
            <div
              className="recipeList__link recipeList__hidden-2x2"
/>,
            Array [
              <Link
                className="recipeList__link recipelist--overlay"
                replace={false}
                to="/recipes/1"
>
                <div
                                className="recipeList__link--container"
                                style={
                                                Object {
                                                                "backgroundImage": "linear-gradient(rgba(255,255,255,0.1), rgba(255,255,255,0.1)), url()",
                                                              }
                                }
                >
                                <div
                                                className="recipeList__link--innercontainer"
                                >
                                                <h2
                                                                className="recipeList__recipeTitle"
                                                />
                                </div>
                </div>
</Link>,
              <Link
                className="recipeList__link recipelist--overlay"
                replace={false}
                to="/recipes/2"
>
                <div
                                className="recipeList__link--container"
                                style={
                                                Object {
                                                                "backgroundImage": "linear-gradient(rgba(255,255,255,0.1), rgba(255,255,255,0.1)), url()",
                                                              }
                                }
                >
                                <div
                                                className="recipeList__link--innercontainer"
                                >
                                                <h2
                                                                className="recipeList__recipeTitle"
                                                />
                                </div>
                </div>
</Link>,
            ],
            Array [
              <div
                className="recipeList__link"
/>,
            ],
          ],
          "className": "recipeList__allrecipes",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1
                className="recipeList__title"
>
                My Recipes
</h1>,
              "className": "recipeList__link",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "My Recipes",
                "className": "recipeList__title",
              },
              "ref": null,
              "rendered": "My Recipes",
              "type": "h1",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "recipeList__link",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "recipeList__link recipeList__hidden-2x2",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "children": <div
                className="recipeList__link--container"
                style={
                                Object {
                                                "backgroundImage": "linear-gradient(rgba(255,255,255,0.1), rgba(255,255,255,0.1)), url()",
                                              }
                }
>
                <div
                                className="recipeList__link--innercontainer"
                >
                                <h2
                                                className="recipeList__recipeTitle"
                                />
                </div>
</div>,
              "className": "recipeList__link recipelist--overlay",
              "replace": false,
              "to": "/recipes/1",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="recipeList__link--innercontainer"
>
                  <h2
                                    className="recipeList__recipeTitle"
                  />
</div>,
                "className": "recipeList__link--container",
                "style": Object {
                  "backgroundImage": "linear-gradient(rgba(255,255,255,0.1), rgba(255,255,255,0.1)), url()",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h2
                    className="recipeList__recipeTitle"
/>,
                  "className": "recipeList__link--innercontainer",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "",
                    "className": "recipeList__recipeTitle",
                  },
                  "ref": null,
                  "rendered": "",
                  "type": "h2",
                },
                "type": "div",
              },
              "type": "div",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "children": <div
                className="recipeList__link--container"
                style={
                                Object {
                                                "backgroundImage": "linear-gradient(rgba(255,255,255,0.1), rgba(255,255,255,0.1)), url()",
                                              }
                }
>
                <div
                                className="recipeList__link--innercontainer"
                >
                                <h2
                                                className="recipeList__recipeTitle"
                                />
                </div>
</div>,
              "className": "recipeList__link recipelist--overlay",
              "replace": false,
              "to": "/recipes/2",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="recipeList__link--innercontainer"
>
                  <h2
                                    className="recipeList__recipeTitle"
                  />
</div>,
                "className": "recipeList__link--container",
                "style": Object {
                  "backgroundImage": "linear-gradient(rgba(255,255,255,0.1), rgba(255,255,255,0.1)), url()",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h2
                    className="recipeList__recipeTitle"
/>,
                  "className": "recipeList__link--innercontainer",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "",
                    "className": "recipeList__recipeTitle",
                  },
                  "ref": null,
                  "rendered": "",
                  "type": "h2",
                },
                "type": "div",
              },
              "type": "div",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "className": "recipeList__link",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="recipeList__allrecipes"
>
            <div
                        className="recipeList__link"
            >
                        <h1
                                    className="recipeList__title"
                        >
                                    My Recipes
                        </h1>
            </div>
            <div
                        className="recipeList__link"
            />
            <div
                        className="recipeList__link recipeList__hidden-2x2"
            />
            <Link
                        className="recipeList__link recipelist--overlay"
                        replace={false}
                        to="/recipes/1"
            >
                        <div
                                    className="recipeList__link--container"
                                    style={
                                                Object {
                                                            "backgroundImage": "linear-gradient(rgba(255,255,255,0.1), rgba(255,255,255,0.1)), url()",
                                                          }
                                    }
                        >
                                    <div
                                                className="recipeList__link--innercontainer"
                                    >
                                                <h2
                                                            className="recipeList__recipeTitle"
                                                />
                                    </div>
                        </div>
            </Link>
            <Link
                        className="recipeList__link recipelist--overlay"
                        replace={false}
                        to="/recipes/2"
            >
                        <div
                                    className="recipeList__link--container"
                                    style={
                                                Object {
                                                            "backgroundImage": "linear-gradient(rgba(255,255,255,0.1), rgba(255,255,255,0.1)), url()",
                                                          }
                                    }
                        >
                                    <div
                                                className="recipeList__link--innercontainer"
                                    >
                                                <h2
                                                            className="recipeList__recipeTitle"
                                                />
                                    </div>
                        </div>
            </Link>
            <div
                        className="recipeList__link"
            />
</div>,
          <Connect(Navbar) />,
        ],
        "className": "recipeList__container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="recipeList__link"
>
                <h1
                                className="recipeList__title"
                >
                                My Recipes
                </h1>
</div>,
              <div
                className="recipeList__link"
/>,
              <div
                className="recipeList__link recipeList__hidden-2x2"
/>,
              Array [
                <Link
                  className="recipeList__link recipelist--overlay"
                  replace={false}
                  to="/recipes/1"
>
                  <div
                                    className="recipeList__link--container"
                                    style={
                                                      Object {
                                                                        "backgroundImage": "linear-gradient(rgba(255,255,255,0.1), rgba(255,255,255,0.1)), url()",
                                                                      }
                                    }
                  >
                                    <div
                                                      className="recipeList__link--innercontainer"
                                    >
                                                      <h2
                                                                        className="recipeList__recipeTitle"
                                                      />
                                    </div>
                  </div>
</Link>,
                <Link
                  className="recipeList__link recipelist--overlay"
                  replace={false}
                  to="/recipes/2"
>
                  <div
                                    className="recipeList__link--container"
                                    style={
                                                      Object {
                                                                        "backgroundImage": "linear-gradient(rgba(255,255,255,0.1), rgba(255,255,255,0.1)), url()",
                                                                      }
                                    }
                  >
                                    <div
                                                      className="recipeList__link--innercontainer"
                                    >
                                                      <h2
                                                                        className="recipeList__recipeTitle"
                                                      />
                                    </div>
                  </div>
</Link>,
              ],
              Array [
                <div
                  className="recipeList__link"
/>,
              ],
            ],
            "className": "recipeList__allrecipes",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1
                  className="recipeList__title"
>
                  My Recipes
</h1>,
                "className": "recipeList__link",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "My Recipes",
                  "className": "recipeList__title",
                },
                "ref": null,
                "rendered": "My Recipes",
                "type": "h1",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "recipeList__link",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "recipeList__link recipeList__hidden-2x2",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "class",
              "props": Object {
                "children": <div
                  className="recipeList__link--container"
                  style={
                                    Object {
                                                      "backgroundImage": "linear-gradient(rgba(255,255,255,0.1), rgba(255,255,255,0.1)), url()",
                                                    }
                  }
>
                  <div
                                    className="recipeList__link--innercontainer"
                  >
                                    <h2
                                                      className="recipeList__recipeTitle"
                                    />
                  </div>
</div>,
                "className": "recipeList__link recipelist--overlay",
                "replace": false,
                "to": "/recipes/1",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="recipeList__link--innercontainer"
>
                    <h2
                                        className="recipeList__recipeTitle"
                    />
</div>,
                  "className": "recipeList__link--container",
                  "style": Object {
                    "backgroundImage": "linear-gradient(rgba(255,255,255,0.1), rgba(255,255,255,0.1)), url()",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h2
                      className="recipeList__recipeTitle"
/>,
                    "className": "recipeList__link--innercontainer",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "",
                      "className": "recipeList__recipeTitle",
                    },
                    "ref": null,
                    "rendered": "",
                    "type": "h2",
                  },
                  "type": "div",
                },
                "type": "div",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "class",
              "props": Object {
                "children": <div
                  className="recipeList__link--container"
                  style={
                                    Object {
                                                      "backgroundImage": "linear-gradient(rgba(255,255,255,0.1), rgba(255,255,255,0.1)), url()",
                                                    }
                  }
>
                  <div
                                    className="recipeList__link--innercontainer"
                  >
                                    <h2
                                                      className="recipeList__recipeTitle"
                                    />
                  </div>
</div>,
                "className": "recipeList__link recipelist--overlay",
                "replace": false,
                "to": "/recipes/2",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <div
                    className="recipeList__link--innercontainer"
>
                    <h2
                                        className="recipeList__recipeTitle"
                    />
</div>,
                  "className": "recipeList__link--container",
                  "style": Object {
                    "backgroundImage": "linear-gradient(rgba(255,255,255,0.1), rgba(255,255,255,0.1)), url()",
                  },
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <h2
                      className="recipeList__recipeTitle"
/>,
                    "className": "recipeList__link--innercontainer",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "",
                      "className": "recipeList__recipeTitle",
                    },
                    "ref": null,
                    "rendered": "",
                    "type": "h2",
                  },
                  "type": "div",
                },
                "type": "div",
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "host",
              "props": Object {
                "className": "recipeList__link",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
